src/lib/expression/window_expression.cpp

```diff
@@ -30,9 +46,10 @@ bool FrameBound::operator==(const FrameBound& rhs) const {
 }
 
 size_t FrameBound::hash() const {
-  auto hash_value = offset;
+  auto hash_value = size_t{0};
+  boost::hash_combine(hash_value, offset);
```

Comment:
So this is about `boost::hash_value`.

Excerpts from Boost doc:

>### [hash_value](https://www.boost.org/doc/libs/1_84_0/libs/container_hash/doc/html/hash.html#ref_hash_value)
>
>```c++
>// Enabled only when T is an integral type
>template<class T>
>  std::size_t hash_value( T v );
>```
>
> **Returns:**
>    When the value of `v` fits into `std::size_t`, when `T` is an unsigned type, or into `ssize_t`, when `T` is a signed type, `static_cast<std::size_t>(v)`.
>
>    Otherwise, an unspecified value obtained by mixing the value bits of `v`.
>
> ### [hash_combine](https://www.boost.org/doc/libs/1_84_0/libs/container_hash/doc/html/hash.html#notes_hash_combine)
>```c++
>template<class T>
>void hash_combine(size_t & seed, T const & v)
>{
>    seed ^= hash_value(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
>}
>```
(The magic `0x9e3779b9` is to avoid having a value of 0 for any sequence of 0s combined with an initial value of 0.)

```c++
auto hash_value = size_t{0};
boost::hash_combine(hash_value, offset);
```
--> `hash_value == 0 ^ (0x9e3779b9 + offset)`
Do with that information whatever you want, but it should not make a difference to just use `offset` as initial value (other than saving three CPU cycles or so).

Reply:
I just wanted to stick to the documented way of using `hash_combine`. I don't have objections if you opt for the previous solution.

Reply:
I'm fine with it, just wanted to point out that this will not really change anything, as well as
https://github.com/hyrise/hyrise/blob/37058e1b065710386f9913fef78bbcdf1a283ef1/src/lib/expression/window_function_expression.cpp#L176-L178

is just a slightly obfuscated version of directly writing 
```c++
size_t WindowFunctionExpression::_shallow_hash() const { 
  return static_cast<size_t>(window_function); 
} 
```
(where I hope the compiler optimizes that anyway).

But well, that's all somewhere deep down Boost or the STL and I think it's fair to give good examples on how students should do sth like that without any assumptions.

Reply:
Well, I assume there are no restrictions on stdlib implementations having `size_t-value = hash(size_t-value)`. So it's not guaranteed to be the same.
But that's a good point. The hash would never be any better. Given that there *could* be collisions in the distant future, should we use the actual value with a comment?

Reply:
As discussed offline: I'm fine with using explicit `std::hash`es and `hash_combine` without making any assumptions about the STL/Boost implementation.